Test error on first build: unexpected hash test result for the dst repo

When running `make` for the first time, I hit this:

```shell
go test -race -coverprofile=coverage.out ./...
--- FAIL: TestDoMirror (0.87s)
    git_test.go:577: unexpected hash test result for the dst repo
FAIL
coverage: 85.5% of statements
FAIL	github.com/agherzan/git-mirror-me	1.215s
ok  	github.com/agherzan/git-mirror-me/cmd/git-mirror-me	1.375s	coverage: 80.0% of statements
ok  	github.com/agherzan/git-mirror-me/internal/utils	0.809s	coverage: 82.2% of statements
FAIL
make: *** [test] Error 1
```

Rerunning `make` then succeeds (I guess internally something gets updated?).

If I `make clean` and `make` again, the issue happens again.
This is right after checking out the repo. Running on MacOS Monterey 12.3.1 (21E258)

```shell
$ go version
go version go1.18.1 darwin/amd64
```

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -1,0 +1,19 @@
+# SPDX-FileCopyrightText: Andrei Gherzan <andrei@gherzan.com>
+#
+# SPDX-License-Identifier: MIT
+
+run:
+  tests: true
+
+linters-settings:
+  gocyclo:
+    min-complexity: 15
+
+linters:
+  disable-all: true
+  enable:
+    - gocyclo
+    - gofmt
+    - govet
+    - ineffassign
+    - misspell

--- a/Makefile
+++ b/Makefile
@@ -2,12 +2,9 @@
 #
 # SPDX-License-Identifier: MIT
 
-INEFFASSIGN := $(shell command -v ineffassign 2> /dev/null)
-GOCYCLO := $(shell command -v gocyclo 2> /dev/null)
-MISSPELL := $(shell command -v misspell 2> /dev/null)
 REUSE := $(shell command -v reuse 2> /dev/null)
 
-all: testcover checks build
+all: test testcover lint reuse build
 
 test:
 	go test -race -coverprofile=coverage.out ./...
@@ -15,30 +12,8 @@ test:
 testcover: test
 	go tool cover -func=coverage.out
 
-checks: vet fmt reuse
-ifdef INEFFASSIGN
-	ineffassign ./...
-endif
-ifdef GOCYCLO
-	gocyclo -ignore "_test" -over 15 .
-endif
-ifdef MISSPELL
-	misspell .
-endif
-
-vet:
-	go vet ./...
-
-fmt:
-	@echo "gofmt checks..."; fails="$$(go list -f '{{.Dir}}' ./... | \
-		xargs -L1 gofmt -l | grep -v /vendor/)"; \
-		if [ -n "$$fails" ]; then \
-		for fail in $$fails; do \
-			gofmt -s -d $$fail; \
-		done; \
-		false; \
-	fi;
-	@echo "gofmt checks done."
+lint:
+	golangci-lint run
 
 reuse:
 ifdef REUSE

--- a/README.md
+++ b/README.md
@@ -6,6 +6,11 @@ SPDX-License-Identifier: MIT
 
 # Git Mirror-Me (`GMm`)
 
+[![Go Report Card](https://goreportcard.com/badge/github.com/agherzan/git-mirror-me)](https://goreportcard.com/report/github.com/agherzan/git-mirror-me)
+[![codecov](https://codecov.io/gh/agherzan/git-mirror-me/branch/main/graph/badge.svg?token=O54JVY4W31)](https://codecov.io/gh/agherzan/git-mirror-me)
+[![License](https://img.shields.io/github/license/agherzan/git-mirror-me?label=License)](/COPYING.MIT)
+[![REUSE status](https://api.reuse.software/badge/github.com/agherzan/git-mirror-me)](https://api.reuse.software/info/github.com/agherzan/git-mirror-me)
+
 This CLI tool provides the ability to mirror a repository to any other git
 repository with optional SSH authentication. For example, it can be used with
 repositories on GitHub, GitLab, Bitbucket, etc.

--- a/README.md
+++ b/README.md
@@ -13,6 +13,12 @@ repositories on GitHub, GitLab, Bitbucket, etc.
 Why "Me"? The name derives from the tool's "ability" to default the source
 repository to the value computed from a GitHub action environment.
 
+## Install
+
+```
+go install github.com/agherzan/git-mirror-me/cmd/git-mirror-me@latest
+```
+
 ## Build
 
 Use the provided `make` script.

--- a/README.md
+++ b/README.md
@@ -75,20 +75,14 @@ descending order of their precedence:
 * The hosts public keys used for host validation.
 * The format needs to be based on the`known_hosts` file.
 
-## Tests and Checks
+## Tests and Linters
 
 Use the provided `make` script. For tests, a `tests` target is provided: `make
 tests`.
 
-Extra checks can be executed using with `make checks`. This make target has
-additional but optional host dependencies on:
-
-* [ineffassign](https://github.com/gordonklaus/ineffassign) -  Detect
-  ineffectual assignments in Go code.
-* [gocyclo](https://github.com/fzipp/gocyclo) -  Calculate cyclomatic
-  complexities of functions in Go source code.
-* [misspell](https://github.com/client9/misspell) - Correct commonly misspelled
-  English words in source files.
+Linters can be executed using `make lint`. This make target has additional host
+dependencies on
+[golangci-lint](https://github.com/golangci/golangci-lint).
 
 ## Contributing
 

--- a/cmd/git-mirror-me/args.go
+++ b/cmd/git-mirror-me/args.go
@@ -32,7 +32,7 @@ func parseArgs(progName string, arguments []string) (*mirror.Config, string, err
 CLI arguments/flags
 `, path.Base(progName))
 
-		flag.PrintDefaults()
+		flags.PrintDefaults()
 		fmt.Fprintf(output,
 			`
 Environment variables

--- a/cmd/git-mirror-me/main.go
+++ b/cmd/git-mirror-me/main.go
@@ -16,7 +16,7 @@ import (
 func run(logger *mirror.Logger, env map[string]string, progName string, args []string) error {
 	conf, output, err := parseArgs(progName, args)
 	if errors.Is(err, flag.ErrHelp) {
-		fmt.Fprintf(os.Stderr, output)
+		fmt.Fprintf(logger.GetOutput(), output)
 
 		return nil
 	} else if err != nil {

--- a/cmd/git-mirror-me/main_test.go
+++ b/cmd/git-mirror-me/main_test.go
@@ -107,7 +107,7 @@ func TestRun(t *testing.T) {
 		t.Fatalf("unexpected refs in the dst repo: %s", dstRepoRefs)
 	}
 
-	ok, err := utils.RepoRefsCheckHash(dstRepo, srcHead)
+	ok, err := utils.RepoRefsCheckHash(dstRepo, srcHead, "refs/")
 	if err != nil {
 		t.Fatalf("dst repo hash check failed: %s", err)
 	}

--- a/git.go
+++ b/git.go
@@ -177,6 +177,33 @@ func setupStagingRepo(conf Config, logger *Logger) (*git.Repository, error) {
 // references to the configured destination repository (as a mirror).
 func pushWithAuth(conf Config, logger *Logger, stagingRepo *git.Repository) error {
 	var auth transport.AuthMethod
+
+	// Set up the public host key.
+	//
+	// The host public keys can be provided via both content and path. When
+	// it is provided via content, we need to use a temporary known_hosts
+	// file.
+	knownHostsPath := conf.GetKnownHostsPath()
+
+	if len(conf.SSH.KnownHosts) != 0 {
+		knownHostsFile, err := ioutil.TempFile("/tmp", tmpKnownHostPathPrefix)
+		if err != nil {
+			return fmt.Errorf("error creating known_hosts tmp file: %w", err)
+		}
+
+		defer func() {
+			knownHostsFile.Close()
+			os.Remove(knownHostsFile.Name())
+		}()
+
+		knownHostsPath = knownHostsFile.Name()
+
+		err = os.WriteFile(knownHostsPath, []byte(conf.SSH.KnownHosts), knownHostsPerm)
+		if err != nil {
+			return fmt.Errorf("error writing known_hosts tmp file: %w", err)
+		}
+	}
+
 	// Set up SSH authentication.
 	if len(conf.SSH.PrivateKey) > 0 {
 		logger.Debug(conf.Debug, "Using SSH authentication.")
@@ -184,30 +211,6 @@ func pushWithAuth(conf Config, logger *Logger, stagingRepo *git.Repository) erro
 		sshKeys, err := ssh.NewPublicKeys("git", []byte(conf.SSH.PrivateKey), "")
 		if err != nil {
 			return fmt.Errorf("failed to setup the SSH key: %w", err)
-		}
-
-		// The host public keys can be provided via both content and path. When
-		// it is provided via content, we need to use a temporary known_hosts
-		// file.
-		knownHostsPath := conf.GetKnownHostsPath()
-
-		if len(conf.SSH.KnownHosts) != 0 {
-			knownHostsFile, err := ioutil.TempFile("/tmp", tmpKnownHostPathPrefix)
-			if err != nil {
-				return fmt.Errorf("error creating known_hosts tmp file: %w", err)
-			}
-
-			defer func() {
-				knownHostsFile.Close()
-				os.Remove(knownHostsFile.Name())
-			}()
-
-			knownHostsPath = knownHostsFile.Name()
-
-			err = os.WriteFile(knownHostsPath, []byte(conf.SSH.KnownHosts), knownHostsPerm)
-			if err != nil {
-				return fmt.Errorf("error writing known_hosts tmp file: %w", err)
-			}
 		}
 
 		hostKeyCallback, err := ssh.NewKnownHostsCallback(knownHostsPath)

--- a/git_test.go
+++ b/git_test.go
@@ -34,186 +34,197 @@ Hau0thh3byP4srEz6dAAAADmFuZHJlaUBxd2lya2xlAQIDBA==
 		"AIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl"
 )
 
-// TestFilterOutRefs tests the filterOutRefs function.
-func TestFilterOutRefs(t *testing.T) {
+// TestFilterOutRefsMatch tests the filterOutRefs function when the filter matches
+// some references.
+func TestFilterOutRefsMatch(t *testing.T) {
 	t.Parallel()
 
-	{
-		// Filter matches refs.
-		path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
-		if err != nil {
-			t.Fatalf("failed to create a temporary repo: %s", err)
-		}
-
-		defer os.RemoveAll(path)
-
-		repo, head, err := utils.NewTestRepo(path, []string{
-			"refs/heads/a",
-			"refs/heads/b",
-			"refs/meta/a",
-			"refs/meta/b",
-		})
-		if err != nil {
-			t.Fatalf("failed to create a test repo: %s", err)
-		}
-
-		err = filterOutRefs(repo, []string{"refs/meta"})
-		if err != nil {
-			t.Fatalf("failed to filter refs: %s", err)
-		}
-
-		refs, err := utils.RepoRefsSlice(repo)
-		if err != nil {
-			t.Fatalf("failed to get repo's refs: %s", err)
-		}
-
-		if !utils.SlicesAreEqual(refs, []string{
-			"HEAD",
-			"refs/heads/master",
-			"refs/heads/a",
-			"refs/heads/b",
-		}) {
-			t.Fatalf("unexpected refs in repo: %s", refs)
-		}
-
-		check, err := utils.RepoRefsCheckHash(repo, head)
-		if err != nil {
-			t.Fatal("failed to check repo refs hash")
-		}
-
-		if !check {
-			t.Fatal("unexpected ref hash")
-		}
+	path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
+	if err != nil {
+		t.Fatalf("failed to create a temporary repo: %s", err)
 	}
-	{
-		// Filter doesn't match refs.
-		path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
-		if err != nil {
-			t.Fatalf("failed to create a temporary repo: %s", err)
-		}
 
-		defer os.RemoveAll(path)
+	defer os.RemoveAll(path)
 
-		repo, head, err := utils.NewTestRepo(path, []string{
-			"refs/heads/a",
-			"refs/heads/b",
-			"refs/meta/a",
-			"refs/meta/b",
-		})
-		if err != nil {
-			t.Fatalf("failed to create a test repo: %s", err)
-		}
-
-		err = filterOutRefs(repo, []string{"refs/nonexistent"})
-		if err != nil {
-			t.Fatalf("failed to filter refs: %s", err)
-		}
-
-		refs, err := utils.RepoRefsSlice(repo)
-		if err != nil {
-			t.Fatalf("failed to get repo's refs: %s", err)
-		}
-
-		if !utils.SlicesAreEqual(refs, []string{
-			"HEAD",
-			"refs/heads/master",
-			"refs/heads/a",
-			"refs/heads/b",
-			"refs/meta/a",
-			"refs/meta/b",
-		}) {
-			t.Fatalf("unexpected refs in repo: %s", refs)
-		}
-
-		check, err := utils.RepoRefsCheckHash(repo, head)
-		if err != nil {
-			t.Fatal("failed to check repo refs hash")
-		}
-
-		if !check {
-			t.Fatal("unexpected ref hash")
-		}
+	repo, head, err := utils.NewTestRepo(path, []string{
+		"refs/heads/a",
+		"refs/heads/b",
+		"refs/meta/a",
+		"refs/meta/b",
+	})
+	if err != nil {
+		t.Fatalf("failed to create a test repo: %s", err)
 	}
-	{
-		// Delete all refs.
-		path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
-		if err != nil {
-			t.Fatalf("failed to create a temporary repo: %s", err)
-		}
 
-		defer os.RemoveAll(path)
-
-		repo, head, err := utils.NewTestRepo(path, []string{
-			"refs/heads/a",
-			"refs/heads/b",
-			"refs/meta/a",
-			"refs/meta/b",
-		})
-		if err != nil {
-			t.Fatalf("failed to create a test repo: %s", err)
-		}
-
-		err = filterOutRefs(repo, []string{""})
-		if err != nil {
-			t.Fatalf("failed to filter refs: %s", err)
-		}
-
-		refs, err := utils.RepoRefsSlice(repo)
-		if err != nil {
-			t.Fatalf("failed to get repo's refs: %s", err)
-		}
-		if !utils.SlicesAreEqual(refs, []string{}) {
-			t.Fatalf("unexpected refs in repo: %s", refs)
-		}
-
-		check, err := utils.RepoRefsCheckHash(repo, head)
-		if err != nil {
-			t.Fatal("failed to check repo refs hash")
-		}
-
-		if !check {
-			t.Fatal("unexpected ref hash")
-		}
+	err = filterOutRefs(repo, []string{"refs/meta"})
+	if err != nil {
+		t.Fatalf("failed to filter refs: %s", err)
 	}
-	{
-		// No prefix.
-		path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
-		if err != nil {
-			t.Fatalf("failed to create a temporary repo: %s", err)
-		}
 
-		defer os.RemoveAll(path)
+	refs, err := utils.RepoRefsSlice(repo)
+	if err != nil {
+		t.Fatalf("failed to get repo's refs: %s", err)
+	}
 
-		repo, head, err := utils.NewTestRepo(path, []string{})
-		if err != nil {
-			t.Fatalf("failed to create a test repo: %s", err)
-		}
+	if !utils.SlicesAreEqual(refs, []string{
+		"HEAD",
+		"refs/heads/master",
+		"refs/heads/a",
+		"refs/heads/b",
+	}) {
+		t.Fatalf("unexpected refs in repo: %s", refs)
+	}
 
-		err = filterOutRefs(repo, []string{})
-		if err != nil {
-			t.Fatalf("failed to filter refs: %s", err)
-		}
+	check, err := utils.RepoRefsCheckHash(repo, head)
+	if err != nil {
+		t.Fatal("failed to check repo refs hash")
+	}
 
-		refs, err := utils.RepoRefsSlice(repo)
-		if err != nil {
-			t.Fatalf("failed to get repo's refs: %s", err)
-		}
+	if !check {
+		t.Fatal("unexpected ref hash")
+	}
+}
 
-		if !utils.SlicesAreEqual(refs, []string{
-			"HEAD",
-			"refs/heads/master",
-		}) {
-			t.Fatalf("unexpected refs in repo: %s", refs)
-		}
+// TestFilterOutRefsNoMatch tests the filterOutRefs function when the filter doesn't
+// match.
+func TestFilterOutRefsNoMatch(t *testing.T) {
+	t.Parallel()
 
-		check, err := utils.RepoRefsCheckHash(repo, head)
-		if err != nil {
-			t.Fatal("failed to check repo refs hash")
-		}
+	path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
+	if err != nil {
+		t.Fatalf("failed to create a temporary repo: %s", err)
+	}
 
-		if !check {
-			t.Fatal("unexpected ref hash")
-		}
+	defer os.RemoveAll(path)
+
+	repo, head, err := utils.NewTestRepo(path, []string{
+		"refs/heads/a",
+		"refs/heads/b",
+		"refs/meta/a",
+		"refs/meta/b",
+	})
+	if err != nil {
+		t.Fatalf("failed to create a test repo: %s", err)
+	}
+
+	err = filterOutRefs(repo, []string{"refs/nonexistent"})
+	if err != nil {
+		t.Fatalf("failed to filter refs: %s", err)
+	}
+
+	refs, err := utils.RepoRefsSlice(repo)
+	if err != nil {
+		t.Fatalf("failed to get repo's refs: %s", err)
+	}
+
+	if !utils.SlicesAreEqual(refs, []string{
+		"HEAD",
+		"refs/heads/master",
+		"refs/heads/a",
+		"refs/heads/b",
+		"refs/meta/a",
+		"refs/meta/b",
+	}) {
+		t.Fatalf("unexpected refs in repo: %s", refs)
+	}
+
+	check, err := utils.RepoRefsCheckHash(repo, head)
+	if err != nil {
+		t.Fatal("failed to check repo refs hash")
+	}
+
+	if !check {
+		t.Fatal("unexpected ref hash")
+	}
+}
+
+// TestFilterOutRefsDeleteAll tests the filterOutRefs function for deleting all
+// references.
+func TestFilterOutRefsDeleteAll(t *testing.T) {
+	t.Parallel()
+
+	path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
+	if err != nil {
+		t.Fatalf("failed to create a temporary repo: %s", err)
+	}
+
+	defer os.RemoveAll(path)
+
+	repo, head, err := utils.NewTestRepo(path, []string{
+		"refs/heads/a",
+		"refs/heads/b",
+		"refs/meta/a",
+		"refs/meta/b",
+	})
+	if err != nil {
+		t.Fatalf("failed to create a test repo: %s", err)
+	}
+
+	err = filterOutRefs(repo, []string{""})
+	if err != nil {
+		t.Fatalf("failed to filter refs: %s", err)
+	}
+
+	refs, err := utils.RepoRefsSlice(repo)
+	if err != nil {
+		t.Fatalf("failed to get repo's refs: %s", err)
+	}
+
+	if !utils.SlicesAreEqual(refs, []string{}) {
+		t.Fatalf("unexpected refs in repo: %s", refs)
+	}
+
+	check, err := utils.RepoRefsCheckHash(repo, head)
+	if err != nil {
+		t.Fatal("failed to check repo refs hash")
+	}
+
+	if !check {
+		t.Fatal("unexpected ref hash")
+	}
+}
+
+// TestFilterOutRefsNoPrefix tests the filterOutRefs function when there is no prefix
+// provided.
+func TestFilterOutRefsNoPrefix(t *testing.T) {
+	t.Parallel()
+
+	path, err := ioutil.TempDir("/tmp", "git-mirror-me-test-")
+	if err != nil {
+		t.Fatalf("failed to create a temporary repo: %s", err)
+	}
+
+	defer os.RemoveAll(path)
+
+	repo, head, err := utils.NewTestRepo(path, []string{})
+	if err != nil {
+		t.Fatalf("failed to create a test repo: %s", err)
+	}
+
+	err = filterOutRefs(repo, []string{})
+	if err != nil {
+		t.Fatalf("failed to filter refs: %s", err)
+	}
+
+	refs, err := utils.RepoRefsSlice(repo)
+	if err != nil {
+		t.Fatalf("failed to get repo's refs: %s", err)
+	}
+
+	if !utils.SlicesAreEqual(refs, []string{
+		"HEAD",
+		"refs/heads/master",
+	}) {
+		t.Fatalf("unexpected refs in repo: %s", refs)
+	}
+
+	check, err := utils.RepoRefsCheckHash(repo, head)
+	if err != nil {
+		t.Fatal("failed to check repo refs hash")
+	}
+
+	if !check {
+		t.Fatal("unexpected ref hash")
 	}
 }
 

--- a/git_test.go
+++ b/git_test.go
@@ -75,7 +75,7 @@ func TestFilterOutRefsMatch(t *testing.T) {
 		t.Fatalf("unexpected refs in repo: %s", refs)
 	}
 
-	check, err := utils.RepoRefsCheckHash(repo, head)
+	check, err := utils.RepoRefsCheckHash(repo, head, "")
 	if err != nil {
 		t.Fatal("failed to check repo refs hash")
 	}
@@ -128,7 +128,7 @@ func TestFilterOutRefsNoMatch(t *testing.T) {
 		t.Fatalf("unexpected refs in repo: %s", refs)
 	}
 
-	check, err := utils.RepoRefsCheckHash(repo, head)
+	check, err := utils.RepoRefsCheckHash(repo, head, "")
 	if err != nil {
 		t.Fatal("failed to check repo refs hash")
 	}
@@ -174,7 +174,7 @@ func TestFilterOutRefsDeleteAll(t *testing.T) {
 		t.Fatalf("unexpected refs in repo: %s", refs)
 	}
 
-	check, err := utils.RepoRefsCheckHash(repo, head)
+	check, err := utils.RepoRefsCheckHash(repo, head, "")
 	if err != nil {
 		t.Fatal("failed to check repo refs hash")
 	}
@@ -218,7 +218,7 @@ func TestFilterOutRefsNoPrefix(t *testing.T) {
 		t.Fatalf("unexpected refs in repo: %s", refs)
 	}
 
-	check, err := utils.RepoRefsCheckHash(repo, head)
+	check, err := utils.RepoRefsCheckHash(repo, head, "")
 	if err != nil {
 		t.Fatal("failed to check repo refs hash")
 	}
@@ -493,7 +493,7 @@ func TestSetupStagingRepo(t *testing.T) {
 		t.Fatalf("unexpected refs in staging repo: %s", stagingRepoRefs)
 	}
 
-	ok, err := utils.RepoRefsCheckHash(stagingRepo, srcHead)
+	ok, err := utils.RepoRefsCheckHash(stagingRepo, srcHead, "")
 	if err != nil {
 		t.Fatalf("utils.RepoRefsCheckHash failed: %s", err)
 	}
@@ -579,7 +579,7 @@ func TestDoMirror(t *testing.T) {
 		t.Fatalf("unexpected refs in the dst repo: %s", dstRepoRefs)
 	}
 
-	ok, err := utils.RepoRefsCheckHash(dstRepo, srcHead)
+	ok, err := utils.RepoRefsCheckHash(dstRepo, srcHead, "refs/")
 	if err != nil {
 		t.Fatalf("dst repo hash check failed: %s", err)
 	}

--- a/internal/utils/utils.go
+++ b/internal/utils/utils.go
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"fmt"
 	"sort"
+	"strings"
 	"time"
 
 	"github.com/go-git/go-billy/v5/memfs"
@@ -208,9 +209,10 @@ func RefsToStrings(refs []*plumbing.Reference) []string {
 	return str
 }
 
-// RepoRefsCheckHash checks if all the references in a repository point to the
-// same hash.
-func RepoRefsCheckHash(repo *git.Repository, hash plumbing.Hash) (bool, error) {
+// RepoRefsCheckHash checks if the references in a repository point to the same
+// hash. When prefix is provided, only the references prefixed accordingly are
+// checked.
+func RepoRefsCheckHash(repo *git.Repository, hash plumbing.Hash, prefix string) (bool, error) {
 	var result bool
 
 	refs, err := repo.References()
@@ -221,6 +223,9 @@ func RepoRefsCheckHash(repo *git.Repository, hash plumbing.Hash) (bool, error) {
 	result = true
 
 	_ = refs.ForEach(func(ref *plumbing.Reference) error {
+		if len(prefix) != 0 && !strings.HasPrefix(ref.Name().String(), prefix) {
+			return nil
+		}
 		if ref.Hash() != hash {
 			result = false
 		}

--- a/internal/utils/utils_test.go
+++ b/internal/utils/utils_test.go
@@ -169,7 +169,7 @@ func TestNewTestRepo(t *testing.T) {
 		t.Fatalf("unexpected refs in repo: %s", refs)
 	}
 
-	check, err := RepoRefsCheckHash(repo, hash)
+	check, err := RepoRefsCheckHash(repo, hash, "")
 	if err != nil {
 		t.Fatal("failed to check repo refs hash")
 	}
@@ -278,7 +278,7 @@ func TestRepoRefsCheckHash(t *testing.T) {
 		t.Fatalf("error creating a test repo %s", err)
 	}
 
-	pass, err := RepoRefsCheckHash(repo, hash)
+	pass, err := RepoRefsCheckHash(repo, hash, "")
 	if err != nil {
 		t.Fatalf("RepoRefsCheckHash failed: %s", err)
 	}
@@ -287,7 +287,25 @@ func TestRepoRefsCheckHash(t *testing.T) {
 		t.Fatal("unexpected hash test result")
 	}
 
-	pass, err = RepoRefsCheckHash(repo, plumbing.NewHash(""))
+	pass, err = RepoRefsCheckHash(repo, hash, "refs/")
+	if err != nil {
+		t.Fatalf("RepoRefsCheckHash failed: %s", err)
+	}
+
+	if !pass {
+		t.Fatal("unexpected hash test result")
+	}
+
+	pass, err = RepoRefsCheckHash(repo, hash, "nomatch/")
+	if err != nil {
+		t.Fatalf("RepoRefsCheckHash failed: %s", err)
+	}
+
+	if !pass {
+		t.Fatal("unexpected hash test result")
+	}
+
+	pass, err = RepoRefsCheckHash(repo, plumbing.NewHash(""), "")
 	if err != nil {
 		t.Fatalf("RepoRefsCheckHash failed: %s", err)
 	}

--- a/log.go
+++ b/log.go
@@ -20,6 +20,7 @@ type Logger struct {
 	warning *log.Logger
 	err     *log.Logger
 	fatal   *log.Logger
+	output  io.Writer
 }
 
 // NewLogger returns a new Logger that will use the passed io.Writer.
@@ -30,8 +31,15 @@ func NewLogger(out io.Writer) *Logger {
 	logger.warning = log.New(out, "[WARN ]: ", 0)
 	logger.err = log.New(out, "[ERROR]: ", 0)
 	logger.fatal = log.New(out, "[FATAL]: ", 0)
+	logger.output = out
 
 	return &logger
+}
+
+// GetOutput returns the output the logger is using for all the logging
+// operations.
+func (l Logger) GetOutput() io.Writer {
+	return l.output
 }
 
 // Debug is printing a log message using the debug logger when debug mode is

--- a/log_test.go
+++ b/log_test.go
@@ -26,6 +26,14 @@ func TestNewLogger(t *testing.T) {
 	}
 }
 
+// TestGetOutput tests the GetOutput function.
+func TestGetOutput(t *testing.T) {
+	t.Parallel()
+
+	logger := NewLogger(os.Stderr)
+	_ = logger.GetOutput()
+}
+
 // TestDebug checks debug logging.
 func TestDebug(t *testing.T) {
 	t.Parallel()

--- a/log_test.go
+++ b/log_test.go
@@ -125,6 +125,7 @@ func TestFatal(t *testing.T) {
 		return
 	}
 
+	//#nosec G204 -- This is only for testing
 	cmd := exec.Command(os.Args[0], "-test.run=TestFatal")
 	cmd.Env = append(os.Environ(), "RUNTESTFATAL=1")
 

--- a/log_test.go
+++ b/log_test.go
@@ -129,8 +129,8 @@ func TestFatal(t *testing.T) {
 	cmd := exec.Command(os.Args[0], "-test.run=TestFatal")
 	cmd.Env = append(os.Environ(), "RUNTESTFATAL=1")
 
-	err := cmd.Run()
-	if e := (&exec.ExitError{}); errors.As(err, &e) && !e.Success() {
+	e := (&exec.ExitError{})
+	if err := cmd.Run(); errors.As(err, &e) && !e.Success() {
 		return
 	}
 

--- a/log_test.go
+++ b/log_test.go
@@ -96,8 +96,9 @@ func TestError(t *testing.T) {
 
 // TestFatalMock checks fatal logging using a mocked exit function.
 func TestFatalMock(t *testing.T) {
-	t.Parallel()
-
+	// We can't run this in parallel with other tests as it messes up with
+	// os.Exit which other tests might require - for example TestFatal. So
+	// do not flag it with t.Parallel().
 	var buf bytes.Buffer
 	logger := NewLogger(&buf)
 

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -13,7 +13,8 @@ description: |
   repository to the value computed from a GitHub action environment.
 license: MIT
 adopt-info: git-mirror-me 
-base: core20
+base: bare
+build-base: core20
 
 confinement: strict
 
@@ -52,3 +53,4 @@ apps:
   git-mirror-me:
     command: bin/git-mirror-me
     plugs: [network, home]
+    adapter: none

